clustered/src/main/java/consumer/RebootConsumer.java

```diff
@@ -0,0 +1,86 @@
+package consumer;
+
+import static commons.Config.BY;
+import static commons.Config.REBOOT_ADDRESS;
+import static commons.Config.STATUS;
+
+import java.util.UUID;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import io.vertx.core.Future;
+import io.vertx.core.VertxOptions;
+import io.vertx.core.json.JsonObject;
+import io.vertx.reactivex.core.AbstractVerticle;
+import io.vertx.reactivex.core.Vertx;
+import io.vertx.reactivex.core.shareddata.Lock;
+
+public class RebootConsumer extends AbstractVerticle {
+
+   private final Logger logger = Logger.getLogger(RebootConsumer.class.getName());
+   private boolean reboot = false;
+   private String id = "ID-" + UUID.randomUUID().toString().substring(0, 4).toUpperCase();
+
+   @Override
+   public void start(Future<Void> startFuture) throws Exception {
+      logger.info("Reboot verticle " + id + " started");
+      vertx.eventBus().consumer("ids", message -> {
+         int id = ((Integer) message.body()).intValue();
+
+         if (id == 0) {
+            launchReboot();
+         }
+      });
+
+      startFuture.complete();
+   }
+
+   private void launchReboot() {
+      if (!reboot) {
+         vertx.sharedData().getLock("lock", ar -> {
+            if (ar.succeeded()) {
+               Lock lock = ar.result();
+               reboot = true;
+               vertx.eventBus().send(REBOOT_ADDRESS, startRebootMessage());
+               logger.info(">> Start system reboot ... ");
+
+               vertx.setTimer(3000, h -> {
+                  vertx.eventBus().send(REBOOT_ADDRESS, endRebootMessage());
+                  logger.info("<< Reboot Over");
+                  lock.release();
```

Comment:
In this case it is safe to release the lock after updating the `reboot` field, because the verticle is a standard verticle, so there won't be another reboot at the same time. But it's a better practice to switch those instructions

Reply:
ok